internal/services/containers/kubernetes_snapshot_data_source.go

```diff
@@ -0,0 +1,109 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerservice/2023-02-02-preview/snapshots"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type KubernetesSnapshotDataSourceModel struct {
+	Name             string            `tfschema:"name"`
+	ResourceGroup    string            `tfschema:"resource_group_name"`
+	SourceResourceId string            `tfschema:"source_resource_id"`
+	Tags             map[string]string `tfschema:"tags"`
+}
+
+type KubernetesSnapshotDataSource struct{}
+
+var _ sdk.DataSource = KubernetesSnapshotDataSource{}
+
+func (r KubernetesSnapshotDataSource) ResourceType() string {
+	return "azurerm_kubernetes_snapshot"
+}
+
+func (r KubernetesSnapshotDataSource) ModelObject() interface{} {
+	return &KubernetesSnapshotDataSourceModel{}
+}
+
+func (r KubernetesSnapshotDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return snapshots.ValidateSnapshotID
+}
+
+func (r KubernetesSnapshotDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+	}
+}
+
+func (r KubernetesSnapshotDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"source_resource_id": {
```

Comment:
This is presumably the node pool ID that the snapshot is based on so can we rename this to 
```suggestion
		"source_node_pool_id": {
```

Reply:
This could also be the cluster's ID, so maybe we keep it as `source_resource_id`?